driver/src/it/scala/org/mongodb/scala/DocumentationExampleSpec.scala

```diff
@@ -0,0 +1,516 @@
+/*
+ * Copyright 2017 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mongodb.scala
+
+import scala.concurrent.Await
+import scala.concurrent.duration.Duration
+import scala.language.implicitConversions
+
+import org.mongodb.scala.bson.{BsonArray, BsonDocument, BsonString}
+
+//scalastyle:off magic.number
+class DocumentationExampleSpec extends RequiresMongoDBISpec {
+
+  // Implicit functions that execute the Observable and return the results
+  val waitDuration = Duration(5, "seconds")
+  implicit class ObservableExecutor[T](observable: Observable[T]) {
+    def execute(): Seq[T] = Await.result(observable, waitDuration)
+  }
+
+  implicit class SingleObservableExecutor[T](observable: SingleObservable[T]) {
+    def execute(): T = Await.result(observable, waitDuration)
+  }
+
+  "The Scala driver" should "be able to insert" in withCollection { collection =>
```

Comment:
Its provided by the test runner - so its based off the class name.  I can't see that it would matter, it just doesn't make sense to always get the collection from the database.